ludwig/trainers/trainer_llm.py

```diff
@@ -373,6 +374,8 @@ def run_evaluation(
         return False
 
 
+###SXK I'd like to help out in adding more use cases around this fine tuning workflow, lets talk more about this
+###as well,one thing I was wondering about is how tightly coupled are we to the trainer and the corresponding backend
```

Comment:
As things go in Ludwig, we are flexible with the following:

Backend: local, ray, horovod, deepspeed
Trainer: local and ray (ray, horovod and deepspeed all run through Ray)